core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -3199,6 +3202,13 @@ object ZIO {
       zio.provideSome[R2](r2 => r2.add(r1))
   }
 
+  final class ProvideSomeLayer[R0 <: Has[_], -R, +E, +A](private val self: ZIO[R, E, A]) extends AnyVal {
+    def apply[E1 >: E, R1 <: Has[_]](
+      layer: ZLayer[R0, E1, R1]
+    )(implicit ev: R0 with R1 <:< R, tagged: Tagged[R0]): ZIO[R0, E1, A] =
+      self.provideSome[R0 with R1](ev).provideLayer[E1, R0, R0 with R1](layer ++ ZLayer.identity[R0])
```

Comment:
If you flip the order of the `++`, you can delete the `Tagged`. Only the RHS of `++` needs the `Tagged`.

Reply:
I tried doing this and then we need `Tagged` for `R1`. Not sure if one is better than the other.

Reply:
Maybe it is better to require a tag for `R1`, the environment we are "providing", so we just prune the part we're providing versus what is leftover?

Reply:
Yes, that's right. The user will generally know the type they are providing, so a tag will already be available for it.